### PR TITLE
Issue 2388: Bookie Http Service Servlet for Embedded Bookie

### PR DESCRIPTION
**Pull request by rudy2steiner**
Descriptions of the changes in this PR:

### Motivation

Expose bookie http service by servlet 

### Changes

*  Introduce Servlet for bookie http service
*  Jetty http server and test bookie http service servlet

Master Issue: #2388

### CODE REVIEW

File: bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.ContextHandlerCollection;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+/**
+ * Jetty based http server
+ **/
+
+public class JettyHttpServer {
```
Comment (eolivelli) [inline bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java:18]: Please move this class to the tests sources.
This is not intended to be used

File: bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/BookieHttpServiceServlet.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+   */
+  HttpServer.Method httpServerMethod(HttpServletRequest request) {
+    switch (request.getMethod()) {
+      case "POST":
+        return HttpServer.Method.POST;
+      case "DELETE":
+        return HttpServer.Method.DELETE;
+      case "PUT":
+        return HttpServer.Method.PUT;
+      default:
```
Comment (eolivelli) [inline bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/BookieHttpServiceServlet.java:113]: Please implement explicitly GET and throw for other cases

**Review by eolivelli — CHANGES_REQUESTED** [review]
Great work
I left a few comments PTAL

File: bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java (hunk 1 above)
Comment (rudy2steiner) [reply bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java:18]: Ok!

**Review by sijie — APPROVED** [review]

**Comment by eolivelli** [comment]
@rudy2steiner it would be cool to merge this patch.
Do you have cycles to address review comments ?

File: bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java (hunk 1 above)
Comment (rudy2steiner) [reply bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/JettyHttpServer.java]: > Please move this class to the tests sources.
> This is not intended to be used

done

File: bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/BookieHttpServiceServlet.java (hunk 2 above)
Comment (rudy2steiner) [reply bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/BookieHttpServiceServlet.java]: > Please implement explicitly GET and throw for other cases

done

**Comment by rudy2steiner** [comment]
> @rudy2steiner it would be cool to merge this patch.
> Do you have cycles to address review comments ?
i have addressed all your comments

**Comment by rudy2steiner** [comment]
@eolivelli  do you mind take your time, help review this work again?

**Review by eolivelli — APPROVED** [review]
Lgtm

@sijie can we include this feature in 4.12?

**Pull request merged**